src/AppInstallerCommonCore/Runtime.cpp

```diff
@@ -269,217 +329,282 @@ namespace AppInstaller::Runtime
         s_runtimePathStateName.emplace(std::move(suitablePathPart));
     }
 
-    std::filesystem::path GetPathTo(PathName path)
+    bool PathDetails::ShouldApplyACL() const
     {
-        std::filesystem::path result;
-        bool create = true;
+        // Could be expanded to actually check the current owner/ACL on the path, but isn't worth it currently
+        return (CurrentUser != ACEPermissions::None || Admins != ACEPermissions::None);
+    }
 
-#ifndef WINGET_DISABLE_FOR_FUZZING
-        if (IsRunningInPackagedContext())
+    void PathDetails::ApplyACL() const
+    {
+        ULONG entries = 0;
+        EXPLICIT_ACCESS_W explicitAccess[2];
+
+        decltype(wil::get_token_information<TOKEN_USER>()) userToken;
+        auto adminSID = wil::make_static_sid(SECURITY_NT_AUTHORITY, SECURITY_BUILTIN_DOMAIN_RID, DOMAIN_ALIAS_RID_ADMINS);
+        PSID ownerSID = nullptr;
+
+        if (CurrentUser != ACEPermissions::None)
         {
-            auto appStorage = winrt::Windows::Storage::ApplicationData::Current();
+            userToken = wil::get_token_information<TOKEN_USER>();
 
-            switch (path)
+            if (CurrentUser == ACEPermissions::Owner)
             {
-            case PathName::Temp:
+                ownerSID = userToken->User.Sid;
+            }
+
+            EXPLICIT_ACCESS_W& entry = explicitAccess[entries++];
+            entry = {};
+
+            entry.grfAccessPermissions = AccessPermissionsFrom(CurrentUser);
+            entry.grfAccessMode = SET_ACCESS;
+            entry.grfInheritance = CONTAINER_INHERIT_ACE | OBJECT_INHERIT_ACE;
+
+            entry.Trustee.pMultipleTrustee = nullptr;
+            entry.Trustee.MultipleTrusteeOperation = NO_MULTIPLE_TRUSTEE;
+            entry.Trustee.TrusteeForm = TRUSTEE_IS_SID;
+            entry.Trustee.TrusteeType = TRUSTEE_IS_USER;
+            entry.Trustee.ptstrName = reinterpret_cast<LPWCH>(userToken->User.Sid);
+        }
+
+        if (Admins != ACEPermissions::None)
+        {
+            if (Admins == ACEPermissions::Owner)
             {
-                result = GetPathToUserTemp();
-                result /= s_DefaultTempDirectory;
+                ownerSID = adminSID.get();

```

Comment:
>ownerSID = adminSID.get(); [](http://example.com/codeflow?start=16&length=26)

Do we allow both User and Admins to declare owner? Looks like if both are declared owner, the Admin one will override CurrentUser one. Is this intentional or we should return e_invalidarg for such input?